ResearchKit/Common/ORKWaitStepView.m

```diff
@@ -0,0 +1,153 @@
+                                                   views:NSDictionaryOfVariableBindings(_progressView, _textLabel)]];
+        
+    } else if (_activityIndicatorView) {
+        
+        [constraints addObject:
+         [NSLayoutConstraint constraintWithItem:_activityIndicatorView
+                                      attribute:NSLayoutAttributeCenterX
+                                      relatedBy:NSLayoutRelationEqual
+                                         toItem:self
+                                      attribute:NSLayoutAttributeCenterX
+                                     multiplier:1.0
+                                       constant:0.0]];
+        [constraints addObjectsFromArray:
+         [NSLayoutConstraint constraintsWithVisualFormat:@"H:|-(>=0)-[_activityIndicatorView]-(>=0)-|"
+                                                 options:NSLayoutFormatAlignAllBaseline
+                                                 metrics:nil
+                                                   views:NSDictionaryOfVariableBindings(_activityIndicatorView)]];
+        [constraints addObjectsFromArray:
+         [NSLayoutConstraint constraintsWithVisualFormat:@"V:[_textLabel]-[_activityIndicatorView]|"
+                                                 options:NSLayoutFormatAlignAllCenterX
+                                                 metrics:nil
+                                                   views:NSDictionaryOfVariableBindings(_activityIndicatorView, _textLabel)]];
+    }
+    
+    [NSLayoutConstraint activateConstraints:constraints];
+}
+
+- (void)setIndicatorMask:(ORKProgressIndicatorMask)indicatorMask {
+    if (_indicatorMask != indicatorMask) {
+        _indicatorMask = indicatorMask;
+        
+        switch (_indicatorMask) {
+            case ORKProgressIndicatorMaskProgressBar:
+                [_activityIndicatorView removeFromSuperview];
+                _activityIndicatorView = nil;
+                _progressView = [UIProgressView new];
+                _progressView.translatesAutoresizingMaskIntoConstraints = NO;
+                [self addSubview:_progressView];
+                break;
+            case ORKProgressIndicatorMaskIndeterminate:
+                [_progressView removeFromSuperview];
+                _progressView = nil;
+                _activityIndicatorView = [[ORKProgressView alloc] init];
+                _activityIndicatorView.translatesAutoresizingMaskIntoConstraints = NO;
+                [self addSubview:_activityIndicatorView];
+                break;
+            default:
+                break;
+        }
+        
+        [self setUpConstraints];
+        [self setNeedsUpdateConstraints];
+    }
+}
+
+#pragma mark Accessibility
+
+- (BOOL)isAccessibilityElement {
+    return YES;
+}
+
+- (NSString *)accessibilityLabel {
+    if (_progressView) {
+        NSString *percentage = [NSString stringWithFormat:@"%li%%", (NSInteger)(_progressView.progress * 100.0)];
```

Comment:
Better way to print percentage:
https://developer.apple.com/library/mac/documentation/Cocoa/Conceptual/DataFormatting/Articles/dfNumberFormatting10_4.html#//apple_ref/doc/uid/TP40002368-SW6